storage/innobase/xtrabackup/src/xtrabackup.cc

```diff
@@ -7401,6 +7426,15 @@ void setup_error_messages() {
   my_default_lc_messages->errmsgs->read_texts();
 }
 
+void xb_set_plugin_dir()
+{
+  if (opt_xtra_plugin_dir != NULL) {
+    strncpy(opt_plugin_dir, opt_xtra_plugin_dir, FN_REFLEN);
+  } else {
+    strcpy(opt_plugin_dir, PLUGINDIR);
+  }
+}
+
```

Comment:
is plugin initializing`-+

Reply:
We used this at plugin_init at - https://github.com/percona/percona-xtrabackup/blob/8.0/storage/innobase/xtrabackup/src/keyring_plugins.cc#L76 . Since we will also need this at compoenent to make it read the component_keyring_file.so from the correct directory, I've made this code a function and I call it at https://github.com/altmannmarcelo/percona-xtrabackup/blob/8.0-PXB-2468/storage/innobase/xtrabackup/src/xtrabackup.cc#L7487 so the pluging dir is already set when we init components and plugins.